### PR TITLE
build: Migrate to go mod

### PR DESCRIPTION
**Pull request by loeliger**
Implement the "go mod" mechanism and eliminate the vendor/ subdirectory.

### CODE REVIEW

**Comment by ondrej-fabry** [comment]
I believe this should also remove `Gopkg.toml`/`Gopkg.lock`.

**Comment by rewenset** [comment]
Looks like there is no VPP in resulting image
```
------------------------------------------------------------
 No socket file found at: /run/vpp/api.sock
 VPP binary API socket file is missing!

  - is VPP running with socket for binapi enabled?
  - is the correct socket name configured?

 To enable it add following section to your VPP config:
   socksvr {
     default
   }
------------------------------------------------------------
```

File: docker/dev/Dockerfile [hunk 1]
```diff
@@ -34,7 +44,7 @@ RUN cp -r plugins/vpp/binapi /tmp/orig_binapi && \
  		[ -n "$SKIP_CHECK" ] && exit 0; \
  	}
 
-FROM vppimg AS devimg
```
Comment (ondrej-fabry) [inline docker/dev/Dockerfile]: `FROM vppimg` is what provides VPP to resulting image.

**Comment by loeliger** [comment]
Series reworked to address issues.

**Pull request closed without merging**